### PR TITLE
gh-100374: Fixed a bug in socket.getfqdn()

### PR DESCRIPTION
**Pull request by YuriNek0**
When getfqdn called with name "::", instead of returning gethostname(), it will call gethostbyaddr("::"). This will raise an exception "socket.herror: [Errno 1] Unknown host", which will cause a 30 seconds(timeout) delay and return incorrect result. The solution is to add a filter to the first if statement, in line 792 socket.py.

* Issue: gh-100374

### CODE REVIEW

**Review by corona10 — CHANGES_REQUESTED** [review]
Would you like to add the test for this case?

**Comment by MTAwsl** [comment]
Added test cases in Lib/test/test_socket.py.

I have made the requested changes; please review again

File: Lib/socket.py [hunk 1]
```diff
@@ -785,11 +785,11 @@ def getfqdn(name=''):
 
     First the hostname returned by gethostbyaddr() is checked, then
     possibly existing aliases. In case no FQDN is available and `name`
-    was given, it is returned unchanged. If `name` was empty or '0.0.0.0',
+    was given, it is returned unchanged. If `name` was empty, '0.0.0.0' or '::',
     hostname from gethostname() is returned.
     """
     name = name.strip()
-    if not name or name == '0.0.0.0':
+    if not name or name == '0.0.0.0' or name == '::':
```
Comment (corona10) [inline Lib/socket.py]:
```suggestion
    if not name or name in ('0.0.0.0', '::'):
```

File: Misc/NEWS.d/next/Core and Builtins/2022-12-20-16-14-19.gh-issue-100374.YRrVHT.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Fixed incorrect result and delays in socket.getfqdn. (Patched by Dominic Socular @MTAwsl)
```
Comment (corona10) [inline Misc/NEWS.d/next/Core and Builtins/2022-12-20-16-14-19.gh-issue-100374.YRrVHT.rst]:
```suggestion
Fix incorrect result and delay in :func:`socket.getfqdn`. Patch by Dominic Socular.
```

**Review by corona10 — CHANGES_REQUESTED** [review]
I left nit comment.

**Comment by MTAwsl** [comment]
Thanks for your commit :)

I have made the requested changes; please review again

**Review by corona10 — APPROVED** [review]
lgtm

**Pull request merged**